src/prommis/nanofiltration/costing/diafiltration_cost_model.py

```diff
@@ -222,6 +190,69 @@ def membrane_area_equation(blk):
                 (membrane_length * membrane_width), to_units=units.m**2
             )
 
+        @blk.Constraint()
+        def capital_cost_constraint(blk):
+            return blk.capital_cost == units.convert(
+                (blk.membrane_cost * blk.membrane_area),
+                to_units=blk.costing_package.base_currency,
+            )
+
+        @blk.Constraint()
+        def fixed_operating_cost_constraint(blk):
+            return blk.fixed_operating_cost == units.convert(
+                (
+                    blk.factor_membrane_replacement
+                    * blk.membrane_cost
+                    * blk.membrane_area
+                ),
+                to_units=blk.costing_package.base_currency
+                / blk.costing_package.base_period,
+            )
+
+    def cost_membrane_pressure_drop(
```

Comment:
It looks like the membrane and cascade "shell" are separate units now, where the membrane has material costs (fixed OPEX) and no power usage, and the cascade has power usage (variable OPEX) but no material costs. Am I understanding that correctly?

Reply:
Also, should the cascade have an associated CAPEX?

Reply:
Yes. After updating how the global parameters were defined, I removed the arbitrary `UnitModelBlock` of `m.fs.membrane`, and instead created `UnitModelCostingBlock`s for each membrane stage. However, the membrane pressure drop calculation (now attached to `m.fs.cascade.costing`) is really intended to consider the overall system. This pressure drop is very small for stage 1 and 2. 

And yes, I believe we will want to include material cost for cascade construction. I have to investigate whether a standard method for this exists.

Reply:
If it's a vessel shell, you can see if the IDAES reactor vessel method is appropriate for the size and shape of your cascade: https://github.com/IDAES/idaes-pse/blob/fe422c006a4b2e03b9b2b1a6ba39eba0043380d1/idaes/models/costing/SSLW.py#L431

Reply:
Actually, after looking into this more, it seems like the cost/area of the membranes (accounted for in the capital cost of the stages) includes the whole spiral wound membrane module. Typically, I believe the membrane cascades are "open" and there wouldn't be one large vessel shell around all of the modules. I think we can omit `m.fs.cascade.costing.capital_cost` at least for now.